app/controllers/account/companies_controller.rb

```diff
@@ -0,0 +1,55 @@
+module Account
+  class CompaniesController < ApplicationController
+    before_action :authenticate_user!
+
+    def index
+      @companies = Company.all
+    end
+
+    def show
+      @company = current_company
+    end
+
+    def new
+      @company = Company.new
+    end
+
+    def create
+      @company = current_user.companies.build(company_params)
+      if @company.save
+        redirect_to account_companies_path
+      else
+        render :new
+      end
+    end
+
+    def edit
+      @company = current_company
+    end
+
+    def update
+      @company = current_company
+      if @company.update_attributes(company_params)
+        redirect_to account_company_path
```

Comment:
is this correct path? i believe for resources it should be either `account_companies_path` or `account_company_path(@company)`

Reply:
some flash message?

Reply:
'account_company_path' works for updating but doesn't for creating, so there i'll use 'account_company_path(@company)' as suggested.
will add flashes, thanks